pkgs/by-name/gh/ghciwatch/package.nix

```
@@ -0,0 +1,45 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, haskellPackages
+, nix-update-script
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ghciwatch";
+  version = "0.5.10";
+
+  src = fetchFromGitHub {
+    owner = "MercuryTechnologies";
+    repo = "ghciwatch";
+    rev = "v${version}";
+    hash = "sha256-6afUHLPrSWhgN5LA346tAZ1+gROr+i/ZyCNVnyCd5Tc=";
+  };
+
+  cargoHash = "sha256-og7S3W+DCBlFIvKLZghLT+msBLnS1o7Rea7v2VPsDYA=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.CoreFoundation
+    darwin.apple_sdk.frameworks.CoreServices
+  ];
+
+  GHC_VERSIONS = haskellPackages.ghc.version;
+
+  nativeBuildInputs = with haskellPackages; [
+    ghc
+    cabal-install
+    hpack
+  ];
```

```suggestion
  doCheck = false;
```

The integration test suite here is _really_ heavy -- it launches something like 200 `ghciwatch`+`ghci` sessions, and it works by waiting for log messages so it can be a little bit flaky (e.g. https://github.com/MercuryTechnologies/ghciwatch/pull/219 upstream). As a result, it adds a lot of time and compute to the build (on my machine, 3.3 minutes with tests, 0.3 minutes without tests).

If you want to run the tests here, that's OK, but it may cause you problems in the future (e.g. in upstream CI we run each test up to 3 times via `cargo-nextest`) and I wouldn't recommend it.

---

You could also skip the integration test suite (and these dependencies) by adding `checkFlags = "--lib";` instead.

---

I'm fine with removing them as soon as they start causing issues; for now they do work so why have less checks when we can have more ;) 

Thank you for building this!

---

ah I didn't read it's flakey as well... hmmm, that changes it; do we want to risk it or should I really just remove it?

---

yeah okay, I'll remove it

---

mmmh; `checkFlags = [ "--lib" ]` doesn't work, unfortunately...

---

maybe what I have now is fine as well? :eyes: